app/src/main/java/com/example/recipeopedia/FavoriteRecipeAdapter.java

```diff
@@ -0,0 +1,80 @@
+package com.example.recipeopedia;
+
+import android.content.Intent;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import androidx.annotation.NonNull;
+import androidx.databinding.DataBindingUtil;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.example.recipeopedia.activities.FavoriteRecipeDetailsActivity;
+import com.example.recipeopedia.activities.RecipeDetailsActivity;
+import com.example.recipeopedia.databinding.ItemFavoriteRecipeBinding;
+import com.example.recipeopedia.models.FavoriteRecipe;
+
+import org.parceler.Parcels;
+
+import java.util.List;
+
+public class FavoriteRecipeAdapter extends RecyclerView.Adapter<FavoriteRecipeAdapter.ViewHolder> {
+    public static final String TAG = "FavoriteRecipeAdapter";
+    private List<FavoriteRecipe> mFavoriteRecipes;
+
+    public FavoriteRecipeAdapter(List<FavoriteRecipe> favoriteRecipes) {
+        this.mFavoriteRecipes = favoriteRecipes;
+    }
+
+    @NonNull
+    @Override
+    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType)
+    {
+        Log.d(TAG, "onCreateViewHolder");
+        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
+        View recipeView = inflater.inflate(R.layout.item_favorite_recipe, parent, false);
+        return new ViewHolder(recipeView);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull FavoriteRecipeAdapter.ViewHolder holder, int position)
+    {
+        Log.d(TAG, "onBindViewHolder " + position);
+        FavoriteRecipe favoriteRecipe = mFavoriteRecipes.get(position);
+        holder.bind(favoriteRecipe);
+    }
+
+    @Override
+    public int getItemCount() {
+        return mFavoriteRecipes.size();
+    }
+
+    public class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener
+    {
+        private ItemFavoriteRecipeBinding binding;
+        public ViewHolder(@NonNull View itemView)
+        {
+            super(itemView);
+            binding = DataBindingUtil.bind(itemView);
+            itemView.setOnClickListener(this);
+        }
+
+        public void bind(FavoriteRecipe favoriteRecipe)
+        {
+            binding.setFavoriteRecipe(favoriteRecipe);
+        }
+
+        @Override
+        public void onClick(View v) {
+            int position = getAdapterPosition();
+            if (position != RecyclerView.NO_POSITION)
+            {
+                FavoriteRecipe favoriteRecipe = mFavoriteRecipes.get(position);
+                Intent intent = new Intent(v.getContext(), FavoriteRecipeDetailsActivity.class);
+                intent.putExtra(FavoriteRecipe.class.getSimpleName(), Parcels.wrap(favoriteRecipe));
```

Comment:
Be careful using `getSimpleName()` with anything but a debugging log tag, as the class name can get [obfuscated in release builds](https://developer.android.com/studio/build/shrink-code#obfuscate). So you may end up with an app which works fine until you cut a release build and then it crashes only in prod. I'd be inclined just to use a static final String key.